db/migrate/20170712171513_sites_cultivars_uniqueness.rb

```diff
@@ -1,5 +1,19 @@
 class SitesCultivarsUniqueness < ActiveRecord::Migration
-  def change
-    add_index :sites_cultivars, :site_id, unique: true
+  def up
+    this_hostid = Machine.new.hostid
+
+    execute %{
+      SELECT setval('sites_cultivars_id_seq', GREATEST(1, CAST(1e9 * #{this_hostid}::int AS bigint)), FALSE);
+
+      ALTER TABLE sites_cultivars ADD CONSTRAINT unique_site_id UNIQUE (site_id);
```

Comment:
The `sites_cultivars` table is only for sites that uniquely determine a cultivar.  Right now, this table is empty in `ebi_production`, so adding this constraint won't break anything there.

If we were to allow multiple rows with the same `site_id` value but different `cultivar_id` values, then the trigger functions that automatically determine and set the cultivar associated with a trait based on the trait's site would have no way to operate.  The most it would seem that having such a table could give us would be the ability restrict the possible cultivars for a trait based on its site.

We did talk about making site-cultivar associations time-sensitive so that the combination of a site and date (rather than the site alone) would uniquely determine a cultivar, but we decided simply to use different "sites" (perhaps encompassing exactly the same geographical area) if the cultivar planted in a particular spot changes over time.

Again, for traits associated with sites _not_ in the `sites_cultivars` table, there is no restriction on what cultivar the trait is associated with.